src/format-files.js

```diff
@@ -179,17 +180,25 @@ function formatFile(filePath, prettierESLintOptions, cliOptions) {
   let format$ = rxReadFile(filePath, 'utf8').map(text => {
     fileInfo.text = text
     fileInfo.formatted = format({text, filePath, ...prettierESLintOptions})
+    fileInfo.unchanged = fileInfo.text === fileInfo.formatted
     return fileInfo
   })
 
   if (cliOptions.write) {
     format$ = format$.mergeMap(info => {
-      if (info.text === info.formatted) {
-        return Rx.Observable.of(Object.assign(fileInfo, {unchanged: true}))
+      if (info.unchanged) {
+        return Rx.Observable.of(info)
       } else {
-        return rxWriteFile(filePath, info.formatted).map(() => fileInfo)
+        return rxWriteFile(filePath, info.formatted).map(() => info)
       }
     })
+  } else if (cliOptions.listDifferent) {
+    format$ = format$.map(info => {
+      if (!info.unchanged) {
+        console.log(info.filePath)
```

Comment:
`console.log` is simpler than `process.stdout.write(info.filePath + "\n")` (plus maybe it supports `\r\n` newlines on windows, I didn't try). I don't think there is a drawback...